File: practica 1/sesion 3/DS_S3_DartFilters/Tecnico.dart
import 'Trabajo.dart';

class Tecnico{
  String nombre;
  int edad, experiencia;
  Trabajo trabajo;

  Tecnico(nombre){
    this.nombre = nombre;
  }

  void darParteTrabajo(horas, material, desplazamiento){
    trabajo.darParteTrabajo(horas, material, desplazamiento);
  }

  void cobrar(Trabajo unTrabajo){
    var precioFinal = unTrabajo.getPresupuesto();
    print("El técnico ha cobrado ${precioFinal} €");
  }

  void cobrarCantidad(double cantidad){
    print("*Técnico cobrando ${cantidad} €");
  }

  void asignarTrabajo(Trabajo nuevo_trabajo){
    this.trabajo = nuevo_trabajo;
  }

  void realizarTrabajo(){
    if (trabajo != null) {
      print("*Realizando trabajo");
      trabajo.realizado();
    }
    else{
      print("El técnico no tiene trabajo todavía");
    }
  }

  void darPresupuesto(double presupuesto){
    trabajo.setPresupuesto(presupuesto);
  }

  String getNombre(){return this.nombre;}
}

File: practica 1/sesion 3/DS_S3_C++Filters/Tecnico.cpp
#include <exception>
#include <iostream>
using namespace std;

#include "Tecnico.h"

using namespace std;

Tecnico::Tecnico(string nombre_tecnico) {
	this->nombre = nombre_tecnico;
}

void Tecnico::cobrar(Trabajo unTrabajo) {
	float precioFinal = unTrabajo.getPresupuesto();
	cout << "El técnico ha cobrado " << precioFinal << " €." << endl ;
}

void Tecnico::darPresupuesto(const float & presupuesto){
	trabajo->setPresupuesto(presupuesto);
}

void Tecnico::cobrar(const float cantidad) {
	cout << "*Tecnico cobrando " << cantidad << " euros*" << endl;
}

void Tecnico::darParteTrabajo(
	const int & horas, const int & material, const int & desplazamiento){

	trabajo->darParteTrabajo(horas, material, desplazamiento);
}

string Tecnico::getNombre() { return this->nombre; }

void Tecnico::asignarTrabajo(Trabajo * nuevo_trabajo) {
	this->trabajo = nuevo_trabajo;
}

void Tecnico::realizarTrabajo(){
	if (trabajo) {
		cout << "*realizando trabajo*" << endl;
		trabajo->realizado();
	} else {
		cout << "tecnico no tiene ningun trabajo todavia" << endl;
	}
}
